hubblestack_nova/stat.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+import logging
+
+import fnmatch
+import yaml
+import os
+import copy
+import salt.utils
+
+from distutils.version import LooseVersion
+
+log = logging.getLogger(__name__)
+
+__tags__ = None
+__data__ = None
+
+
+def __virtual__():
+    if salt.utils.is_windows():
+        return False, 'This audit module only runs on linux'
+    global __tags__
+    global __data__
+    yamldir = os.path.dirname(__file__)
+    if not yamldir:
+        yamldir = '.'
+    __data__ = _get_yaml(yamldir)
+    __tags__ = _get_tags(__data__)
+    return True
+
+
+def audit(tags, verbose=False):
+    '''
+    Run the grep audits contained in the YAML files processed by __virtual__
+    '''
+    ret = {'Success': [], 'Failure': []}
+
+    for tag in __tags__:
+        if fnmatch.fnmatch(tag, tags):
+            for tag_data in __tags__[tag]:
+                name = tag_data['name']
+                expected = {}
+                for e in ['mode', 'user', 'group']:
```

Comment:
Can we add support for `gid` and `uid` as well, here? That way the YAML will support uid, gid, user, group, and mode.

`file.stats` returns all of this information, so it should work if you just add them here.